scripts/symbol_check/readme.md

```diff
@@ -0,0 +1,88 @@
+
+There are two scripts included:
+1. `generate_unsupported_symbol.py`
+   1. A Python script that generates a list of symbols that are present in glibc but not in Mystikos's CRT.
+   2. This script should be used manually to generate file that contains unsupported symbols. Only regenerate the file when necessary.
+2. `check_unsupported_symbol.sh`
+   1. A script that lists the symbols required by a Mystikos app and print out any symbols that are present in the file generated by `generate_unsupported_symbol.py`.
+   2. This script will be called as part of `scripts/appbuilder`.
+
+## Directory structure
+
+```
+symbol-check        # root directory for this feature
+├── check_unsupported_symbol.sh    # checks user program's usage of unsupported symbols
+├── generate_unsupported_symbol.py # generates diff between glibc/mystikos symbols
+├── readme.md       # this readme
+└── symbols         # directory that holds symbol files
+    └── unsupported # contains glibc symbols that are not supported in Mystikos
+```
+
+## Implement Detail
+
+### generate_unsupported_symbol.py
+
+A Python script that generates a list of symbols that are present in glibc but not in Mystikos's CRT. We use `nm` to list the symbols from object files.
+
+This script should be used manually to generate file that contains unsupported symbols. Only regenerate the file when necessary.
+
+Example usage:  
+```bash
+./generate_unsupported_symbol.py \
+--iglibc /lib/x86_64-linux-gnu/libc.so.6 /lib/x86_64-linux-gnu/libm.so.6 /lib/x86_64-linux-gnu/libpthread.so.0 \
+--imystikos ../../build/lib/libmystcrt.so \
+-o symbols/unsupported
+```
+
+How it works:
+1. Find the difference between available symbols in Mystikos's CRT (musl) and glibc. We combine the output from 3 glibc object files to build the glibc family symbols  
+
+    1. `/lib/x86_64-linux-gnu/libc.so.6`
+    2. `/lib/x86_64-linux-gnu/libm.so.6`
+    3. `/lib/x86_64-linux-gnu/libpthread.so.0`
+
+    > Example command: `nm -D /lib/x86_64-linux-gnu/libc.so.6`
+
+2. Then we grab the symbols in Mystikos CRT
+   1. `nm ./build/lib/libmystcrt.so`
+
+3. Lastly we generate symbols that are in glibc but not in Mystikos CRT, and write it to file `symbols/unsupported`
+
+    ```python
+    set_unsupported_symbols_in_mystikos = set(symbols_in_glibc_family) - set(symbols_in_mystikos_crt)
+    ```
+
+### check_unsupported_symbol.sh
+
+A script that lists the symbols required by a Mystikos app and print out any symbols that are present in the file generated in part 1.
+
+This script will be called as part of `scripts/appbuilder`.
+
+Example usage:  
+```bash
+./check_unsupported_symbol.sh [docker image] [path to appdir] ./symbols/unsupported
+```
+
+This part will check symbols used in the generated executable file under `appdir/`, and report any symbols that are present in file that contains unsupported symbols.
+
+The binary file to be checked is whatever provided to Dockerfile's `ENTRYPOINT` command by using
```

Comment:
What if the docker image was built without a specific `ENTRYPOINT`? 

Should the tool take an additional parameter that is the PATH to the entry point?

Reply:
Adding optional arguments in case Dockerfile doesn't have ENTRYPOINT.